File: TwoSum.java
import java.util.HashMap;
import java.util.Map;

class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for(int i=0; i<nums.length; i++){
            int complementary = target - nums[i];
            if(map.containsKey(complementary)){
                return new int[] {map.get(complementary),i};
            }
            map.put(nums[i], i);
        }
        return null;
    }
}

File: TwoSum.py
class TwoSum:
    def two_sum(self, nums, target):
        num_set = set()
        for i, num in enumerate(nums):
            complementary = target - num
            if complementary in num_set:
                return [nums.index(complementary),i]
            num_set.add(num)
        return None